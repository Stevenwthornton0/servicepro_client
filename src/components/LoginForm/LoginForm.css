.loginForm {
    margin-left: auto;
    margin-right: auto;
    margin-bottom: 120px;
    width: 25%;
    display: flex;
    flex-direction: column;
    margin-top: 90px;
}

.loginForm > div {
    margin-bottom: 30px;
    margin-left: 0;
}

.loginForm > div > input {
    margin-top: 4px;
    border-radius: 4px;
    border: 1px solid #bdbdbd;
    font-size: 18px;
    padding: 8px;
}

.loginForm > div > input:focus {
    outline: none;
}

.loginButton {
    margin-top: 10px;
    margin-left: 0;
    border: 1px solid lightblue;
    border-radius: 4px;
    font-size: 18px;
    font-weight: 400;
    padding-top: 10px;
    padding-bottom: 10px;
    background-color: lightblue;
}

.loginButton:hover {
    background-color: #424242;
    border: 1px solid #424242;
    color: lightblue;
    cursor: pointer;
}

.loginButton:focus {
    outline: none;
}

.username, .password {
    display: flex;
    flex-direction: column;
}

@media screen and (max-width: 500px) {
    .loginForm {
        width: 60%;
    }
}
