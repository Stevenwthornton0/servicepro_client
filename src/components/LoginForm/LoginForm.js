import React, { Component } from 'react';
import AuthApiService from '../../services/auth-api-service';
import TokenService from '../../services/token-service';
import ServicesContext from '../../contexts/ServicesContext';
import './LoginForm.css';

class LoginForm extends Component {
    static contextType = ServicesContext;

    constructor(props) {
        super(props)
        this.state = {
            error: null,
            waiting: false
        }
    }

    static defaultProps = {
        onLoginSuccess: () => {},
    }

    handleSubmitJwtAuth = ev => {
        ev.preventDefault()
        this.setState({ 
            error: null,
            waiting: true,
        })
        const { user_name, password } = ev.target;

        AuthApiService.postLogin({
            user_name: user_name.value,
            password: password.value
        })
            .then(res => {
                TokenService.saveUsername(user_name.value)
                user_name.value = ''
                password.value = ''
                TokenService.saveAuthToken(res.authToken)
                this.setState({ waiting: false })
                this.props.onLoginSuccess()
            })
            .catch(res => {
                this.setState({ 
                    error: res.error,
                    waiting: false
                })
            })
    }

    render() {
        const { error } = this.state;
        return (
            <form 
                className='loginForm'
                onSubmit={this.handleSubmitJwtAuth}
            >
                <div role='alert'>
                    {error && <p className='error'>{error}</p>}
                </div>
                
                <div className='username'>
                    <label htmlFor='LoginForm_username'>
                        Username
                    </label>
                    <input
                        className='username_input'
                        required
                        name='user_name'
                        id='LoginForm_username'
                    />
                </div>
                <div className='password'>
                    <label htmlFor='LoginForm_password'>
                        Password
                    </label>
                    <input
                        className='password_input'
                        required
                        name='password'
                        type='password'
                        id='LoginForm_passsword'
                    />
                </div>
                <button type='submit' className='loginButton'>
                    Log in
                </button>
            </form>
        )
    }
}

export default LoginForm;