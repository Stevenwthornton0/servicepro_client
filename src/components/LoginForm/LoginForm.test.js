import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme'
import LoginForm from './LoginForm';

Enzyme.configure({ adapter: new Adapter() })

describe('LoginForm component', () => {
    it('renders without crashing', () => {
        const wrapper = shallow(<LoginForm />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})