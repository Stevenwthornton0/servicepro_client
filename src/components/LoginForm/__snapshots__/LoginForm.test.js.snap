// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component renders without crashing 1`] = `
<form
  className="loginForm"
  onSubmit={[Function]}
>
  <div
    className="red"
    role="alert"
  />
  <div
    className="username"
  >
    <label
      htmlFor="LoginForm_username"
    >
      Username
    </label>
    <input
      className="username_input"
      id="LoginForm_username"
      name="user_name"
      required={true}
    />
  </div>
  <div
    className="password"
  >
    <label
      htmlFor="LoginForm_password"
    >
      Password
    </label>
    <input
      className="password_input"
      id="LoginForm_passsword"
      name="password"
      required={true}
      type="password"
    />
  </div>
  <button
    className="loginButton"
    type="submit"
  >
    Log in
  </button>
  <div
    className="loader"
  />
</form>
`;
